### ISSUE TITLE
Implement nats queue groups

### ISSUE THREAD

**Issue by tonyhb** [google/go-cloud#2753]
It would be good to add support for NATS queue groups for single-subscriber delivery.  There's a comment to this effect (https://github.com/google/go-cloud/blob/master/pubsub/natspubsub/nats.go#L284).

**Comment by tonyhb** [comment]
Semi-related to https://github.com/google/go-cloud/issues/2141.

**Comment by mathantunes** [comment]
Hi,

Any progress on this topic so far ? I'd like to work on it.

**Comment by vangent** [comment]
Nobody is actively working on it. Could you summarize what kind of a change you think this is? I.e., if it's a new constructor and a few lines of code, vs a brand new driver.

**Comment by vangent** [comment]
@mathantunes Nobody is actively working on it. Could you summarize what kind of a change you think this is? I.e., if it's a new constructor and a few lines of code, vs a brand new driver.

**Comment by mathantunes** [comment]
Hi @vangent,

The [NATS QueueSubscribeSync](https://github.com/nats-io/nats.go/blob/76f36eea7e4c57cc5325bb80de40ecede69ffc48/nats.go#L2885) implementation is just a subscribe with a specified queue group and NATS takes care of the message delivery to only one queue subscriber.

* [NATS Docs on QueueSubscriber](https://docs.nats.io/nats-concepts/queue)

On go-cloud, I thing we can **modify the current NATS driver** either by utilizing the  *SubscriptionOptions present [here](https://github.com/google/go-cloud/blob/bb59d59f26d596dcd5eb970cc06572e9da7de7bd/pubsub/natspubsub/nats.go#L285)

Or by creating a new Subscription constructor, as sketched below:
```go
func OpenQueueSubscription(nc *nats.Conn, subject, queue string) (*pubsub.Subscription, error) {
	sub, err = nc.QueueSubscribeSync(subject, queue)
        ds := &subscription{nc, sub}
	return pubsub.NewSubscription(ds, recvBatcherOpts, nil), nil
}
```

**Comment by vangent** [comment]
OK, this sounds like a minor change then; feel free to send a PR. Adding an option to `SubscriberOptions` SGTM.

**Comment by mathantunes** [comment]
Alright, here it goes [pubsub/natspubsub: implement nats queue subscription #2782](https://github.com/google/go-cloud/pull/2782). 

It seems like I can't link the PR to the Issue.

Thank you
